lldb/source/Plugins/SymbolFile/DWARF/DWARFDebugInfoEntry.cpp

```diff
@@ -44,10 +45,20 @@ bool DWARFDebugInfoEntry::Extract(const DWARFDataExtractor &data,
                                   const DWARFUnit *cu,
                                   lldb::offset_t *offset_ptr) {
   m_offset = *offset_ptr;
+  auto report_error = [&](const char *fmt, const auto &...vals) {
+    cu->GetSymbolFileDWARF().GetObjectFile()->GetModule()->ReportError(
```

Comment:
Can we assume `cu` is always a valid pointer? If so, should the signature take it by const reference instead? 

AFAIK only `GetAbbreviationDeclarationPtr` checks that it's not NULL but we have a code path that calls the lambda before as well as when `abbrevDecl` is NULL which would happen if CU is NULL.

Reply:
All of the callers are in DWARFUnit (passing `this`), but I'm not going to pass up an opportunity to add more references. :P